src/s3/activation.ts

```diff
@@ -29,7 +35,53 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
             await downloadFileAsCommand(node)
         }),
         vscode.commands.registerCommand('aws.s3.uploadFile', async (node: S3BucketNode | S3FolderNode) => {
-            await uploadFileCommand(node)
+            const fileLocation = await getFileToUpload()
+            if (!fileLocation) {
+                getLogger().info('UploadFile cancelled')
+                telemetry.recordS3UploadObject({ result: 'Cancelled' })
+                return
+            }
+            const key = node.path + path.basename(fileLocation.fsPath)
+            const bucket:S3.Bucket = { Name: node.bucket.name }
+            await uploadFileCommand(fileLocation, key, bucket, node.s3)
+        }),
+        vscode.commands.registerCommand('aws.s3.uploadFileToS3', async () => {
+            while (true) {
+                const editor = vscode.window.activeTextEditor
+                const document = editor?.document.uri
+                const file = await getFileToUpload(document)
+                if (file) {
+                    try{
+                        const regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1"
+                        const s3Client = ext.toolkitClientBuilder.createS3Client(regionCode) 
+                        
+                        const bucketResponse = await promptUserForBucket(s3Client)
+                        if (bucketResponse === 'back'){
+                            continue
+                        }
+                        if(bucketResponse == 'cancel'){
+                            getLogger().info('No bucket selected, cancelling upload')
+                            telemetry.recordS3UploadObject({ result: 'Cancelled' })
+                            return
+                        }
+                        
+                        const bucket = bucketResponse as S3.Bucket
+                        const key = '' + path.basename(file.fsPath)
+                        return await uploadFileCommand(file, key, bucket, s3Client)
+                    } catch (e) {
+                        
+                        return
```

Comment:
almost always, errors should at least be logged. not silently ignored. However, because of the expanded scope of the `try`, the log message cannot be made precise. As I suggested in https://github.com/aws/aws-toolkit-vscode/pull/1799#discussion_r652225226 , I don't think we should expand the scope of try{} just to avoid `let` @bryceitoc9 .

    getLogger().error('promptUserForBucket failed', e)